packages/react-hooks/src/hooks/progress-state.ts

```diff
@@ -0,0 +1,16 @@
+import {useState} from 'react';
+
+export function useProgressState<T>(
```

Comment:
Is `progress` is the right word here? Typically progress bar APIs use numbers to track changes. Maybe `switch` or `toggle` is more specific to what this is doing?

Reply:
yeah progress can be misleading. `Toggle` or `Switch` might work, how do you feel about `useToggleMethodState`/`useMethodToggleState`? I feel like`useToggleState` could be confused with `useToggle`?

Reply:
`useAsyncActionState`?

Reply:
Thanks for the input at Pat I think `useAsyncActionState` is fairly descriptive, updated the PR!